src/components/burger-constructor/burger-constructor.jsx

```diff
@@ -15,21 +15,33 @@ const BurgerConstructor = () => {
 
   const dispatch = useDispatch();
 
-  const { constructorIngredients, totalPrice } = useSelector(store => store.burger_constructor);
+  const { constructorIngredients } = useSelector(store => store.burger_constructor);
   const { order } = useSelector(store => store.order);
 
-  
+  const { buns, notBuns, disabled, totalPrice } = useMemo(() => {
+    const buns = constructorIngredients.filter(ingredient => ingredient.type === 'bun')
+    const notBuns = constructorIngredients.filter(ingredient => ingredient.type !== 'bun')
+    const disabled = constructorIngredients.length === 0 || buns.length !== 2
```

Comment:
Тут тоже нужно будеть чуток по другому написать.

```suggestion
    const disabled = constructorIngredients.length === 0 || !bun
```